eclair-core/src/main/scala/fr/acinq/eclair/router/Announcements.scala

```diff
@@ -32,20 +32,20 @@ import scala.concurrent.duration._
   */
 object Announcements {
 
-  def channelAnnouncementWitnessEncode(chainHash: ByteVector32, shortChannelId: ShortChannelId, nodeId1: PublicKey, nodeId2: PublicKey, bitcoinKey1: PublicKey, bitcoinKey2: PublicKey, features: ByteVector): ByteVector =
-    sha256(sha256(serializationResult(LightningMessageCodecs.channelAnnouncementWitnessCodec.encode(features :: chainHash :: shortChannelId :: nodeId1 :: nodeId2 :: bitcoinKey1 :: bitcoinKey2 :: HNil))))
+  def channelAnnouncementWitnessEncode(chainHash: ByteVector32, shortChannelId: ShortChannelId, nodeId1: PublicKey, nodeId2: PublicKey, bitcoinKey1: PublicKey, bitcoinKey2: PublicKey, features: ByteVector, unknownFields: ByteVector): ByteVector =
```

Comment:
a bit off-topic, but it feels like inputs to all these `*WitnessEncode` method should be actual gossip messages

Reply:
True, not sure why it's done that way, will fix.

Reply:
I know why: we don't have an object for this, only a codec.